platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java

```diff
@@ -147,11 +149,45 @@ public static void recoverState(
 
         SignedStateFileWriter.writeSignedStateFilesToDirectory(resultingStateDirectory, resultingState);
 
+        updateEmergencyRecoveryFile(resultingStateDirectory, initialState.getConsensusTimestamp());
+
         logger.info(STARTUP.getMarker(), "Recovery process completed");
 
         resultingState.release();
     }
 
+    /**
+     * Update the resulting emergency recovery file to contain the bootstrap timestamp.
+     *
+     * @param recoveryFileDir the directory containing the emergency recovery file
+     * @param bootstrapTime   the consensus timestamp of the bootstrap state
+     */
+    public static void updateEmergencyRecoveryFile(final Path recoveryFileDir, final Instant bootstrapTime) {
+        try {
+            // Read the existing recovery file and write it to a backup directory
+            final EmergencyRecoveryFile oldRecoveryFile = EmergencyRecoveryFile.read(recoveryFileDir);
+            if (oldRecoveryFile == null) {
+                logger.error(EXCEPTION.getMarker(),
+                        "Recovery file does not exist at {}", recoveryFileDir);
+                return;
+            }
+            final Path backupDir = recoveryFileDir.resolve("backup");
+            if (!Files.exists(backupDir)) {
+                Files.createDirectory(backupDir);
```

Comment:
```suggestion
                Files.createDirectories(backupDir);
```
I prefer this approach because it's more robust. Will create entire directory tree, as needed.